server/neptune/workflows/internal/deploy/revision/queue/deployer.go

```diff
@@ -66,16 +67,24 @@ func (p *Deployer) Deploy(ctx workflow.Context, requestedDeployment terraformWor
 	}
 	err = p.TerraformWorkflowRunner.Run(ctx, requestedDeployment)
 
-	// don't wrap this err as it's not necessary and will mess with any err type assertions we might need to do
-	if err != nil {
+	// No need to persist deployment if it's a PlanRejectionError
+	if _, ok := err.(*terraformWorkflow.PlanRejectionError); ok {
 		return nil, err
 	}
+
 	latestDeployment = requestedDeployment.BuildPersistableInfo()
 
-	if err = p.persistLatestDeployment(ctx, latestDeployment); err != nil {
-		logger.Error(ctx, "unable to persist deployment, proceeding with in-memory value")
+	// don't wrap this err as it's not necessary and will mess with any err type assertions we might need to do
+	if persistErr := p.persistLatestDeployment(ctx, latestDeployment); persistErr != nil {
+		logger.Error(ctx, "unable to persist deployment, proceeding with in-memory value", map[string]interface{}{
```

Comment:
`logger.Error(ctx, "unable to persist deployment, proceeding with in-memory value", key.ErrKey, err)`